smartapps/smartthings/hue-connect.src/hue-connect.groovy

```diff
@@ -490,24 +490,25 @@ def ssdpBridgeHandler(evt) {
 		def host = ip + ":" + port
 		log.debug "Device ($parsedEvent.mac) was already found in state with ip = $host."
 		def dstate = bridges."${parsedEvent.ssdpUSN.toString()}"
-		def dni = "${parsedEvent.mac}"
-		def d = getChildDevice(dni)
+		def dniReceived = "${parsedEvent.mac}"
+		def currentDni = dstate.mac
+		def d = getChildDevice(dniReceived)
 		def networkAddress = null
 		if (!d) {
-			childDevices.each {
-				if (it.getDeviceDataByName("mac")) {
-					def newDNI = "${it.getDeviceDataByName("mac")}"
-					d = it
-					if (newDNI != it.deviceNetworkId) {
-						def oldDNI = it.deviceNetworkId
-						log.debug "updating dni for device ${it} with $newDNI - previous DNI = ${it.deviceNetworkId}"
-						it.setDeviceNetworkId("${newDNI}")
-						if (oldDNI == selectedHue) {
-							app.updateSetting("selectedHue", newDNI)
-						}
-						doDeviceSync()
-					}
+			// There might be a mismatch between bridge DNI and the actual bridge mac address, correct that
```

Comment:
I think you're right, but we should make sure that a second bridge wouldn't screw up the first one. Maybe we should be dropping any SSDP messages if the USN doesn't match the one already in the bridge list. Also should we be checking the model name to be equal to "Philips hue bridge" to make sure it's not some other device that uses the same URN?